### PR TITLE
fix schema references

### PR DESCRIPTION
**Pull request by suhailkakar**
**What does this pull request do? Explain your changes. (required)**

The current version of `api-schema.yaml` is having errors on OpenAPI validations and cannot be used by any parser and etc. This PR fixes those errors.

 **Specific updates (required)**
 - moved `object-store-patch-payload `, `stream-set-active-payload`, and `signing-key-response-payload` to `db-schema.yaml`
 - updated `playbackid` to `playbackId` (same as param) 
 - for 4 components removed and `$ref` and instead just added the params

**How did you test each of these updates (required)**
- `yarn compile-schemas` and `yarn test`

**Does this pull request close any open issues?**

**Screenshots (optional)**

**Checklist**

- [X] I have read the **CONTRIBUTING** document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.

### CODE REVIEW

File: packages/api/src/schema/api-schema.yaml [hunk 1]
```diff
@@ -1000,11 +986,10 @@ components:
                   properties:
                     spec:
                       oneOf:
-                        - type: "null"
+                        - type: object
                         - $ref: >-
                             #/components/schemas/asset/properties/storage/properties/ipfs/properties/spec
                 - type: boolean
-                - type: "null"
```
Comment (suhailkakar) [inline packages/api/src/schema/api-schema.yaml]: @victorges can you please confirm if this is correct?

File: packages/api/src/schema/api-schema.yaml [hunk 2]
```diff
@@ -1663,15 +1648,6 @@ components:
         disabled:
           type: boolean
           description: Disable the signing key to allow rotation safely
-    signing-key-response-payload:
```
Comment (victorges) [inline packages/api/src/schema/api-schema.yaml:1666]: This is actually an API object. It should stay here and we should actually change https://github.com/livepeer/livepeer-com/blob/e98108915d4ead04eaec4ba75240084355422d00/packages/api/src/schema/api-schema.yaml#L3466 to reference it instead of the other one which is wrong.

File: packages/api/src/schema/api-schema.yaml [hunk 3]
```diff
@@ -2102,7 +2078,18 @@ components:
...
+                  description: CID of the file on IPFS
+                url:
+                  readOnly: true
+                  type: string
+                  description: URL with IPFS scheme for the file
+                gatewayUrl:
+                  readOnly: true
+                  type: string
+                  description:
+                    URL to access file via HTTP through an IPFS gateway
```
Comment (victorges) [inline packages/api/src/schema/api-schema.yaml]: Actually! I remembered that better than repeating each and every field, we can actually $ref the individual fields and only repeat the list here :)

Can you change the places where you fixed the "inheritance $ref" to be like this instead? suggestion+example:
```suggestion
                cid:
                  $ref: "#/components/schemas/ipfs-file-info/properties/cid"
                url:
                  $ref: "#/components/schemas/ipfs-file-info/properties/url"
                gatewayUrl:
                  $ref: "#/components/schemas/ipfs-file-info/properties/gatewayUrl"
```

File: packages/api/src/schema/api-schema.yaml [hunk 4]
```diff
@@ -2115,8 +2102,23 @@ components:
       type: object
       additionalProperties: false
       properties:
-        $ref: >-
-          #/components/schemas/asset/properties/storage/properties/ipfs/properties/spec/properties
+        nftMetadataTemplate:
```
Comment (victorges) [inline packages/api/src/schema/api-schema.yaml:1906]: Same for refing the fields instead of repeating their specs

File: packages/api/src/schema/db-schema.yaml [hunk 5]
```diff
@@ -1101,3 +1101,37 @@ components:
...
+          type: number
+          description: Timestamp (in milliseconds) at which the stream started.
+          example: 1587667174725
+    signing-key-response-payload:
+      type: object
+      required:
+        - publicKey
+        - privateKey
+      properties:
+        $ref: "#/components/schemas/signing-key/properties"
```
Comment (victorges) [inline packages/api/src/schema/db-schema.yaml]: This will need to change to individual field `$ref` as well

**Review by victorges — APPROVED** [review]
Actually LGTM as the change to $ref individual fields is pretty simple, but let me know if you want another review after you make that change

File: packages/api/src/schema/api-schema.yaml (hunk 3 above)
Comment (suhailkakar) [reply packages/api/src/schema/api-schema.yaml]: Yes, it is way better! updated it!

File: packages/api/src/schema/api-schema.yaml [hunk 6]
```diff
@@ -1000,11 +1000,10 @@ components:
                   properties:
                     spec:
                       oneOf:
-                        - type: "null"
+                        - type: object
```
Comment (victorges) [inline packages/api/src/schema/api-schema.yaml]: Why this change? I think this is changing the schema. Continuing on the other comment.

File: packages/api/src/schema/api-schema.yaml (hunk 1 above)
Comment (victorges) [reply packages/api/src/schema/api-schema.yaml]: This is official JSON schema syntax: https://json-schema.org/understanding-json-schema/reference/null.html

Do you know why does these OpenAPI tools not support it? Either way, we probably need to do more here to actually represent a field that can only be `null`.

**Review by victorges — APPROVED** [review]
Apart from the null thing, LGTM

File: packages/api/src/schema/api-schema.yaml (hunk 6 above)
Comment (suhailkakar) [reply packages/api/src/schema/api-schema.yaml]: I believe with `nullable: true` it should be fine?

File: packages/api/src/schema/api-schema.yaml [hunk 7]
```diff
@@ -995,12 +1005,12 @@ components:
                 pinned files, specify an object with a spec field. False or null
                 means to unpin from IPFS, but it's unsupported right now.
               oneOf:
-                - type: object
+                - type: "null"
```
Comment (victorges) [inline packages/api/src/schema/api-schema.yaml]: I think you over-reverted here haha

File: packages/api/src/schema/api-schema.yaml (hunk 6 above)
Comment (victorges) [reply packages/api/src/schema/api-schema.yaml]: It isn't, because it will start accepting any kind of objects as well not only `null`. We would have to add `nullable` to an existing `oneOf` option here, but the problem is that the only one currently is a `$ref`.

I think there might be a way if we redefined the `spec` object here to tag it as `nullable` (maybe `$ref`'ing the `properties` field instead to avoid too much repetition)

If we decide to ignore these `null` for now, make sure to revert this to `type: "null"`

File: packages/api/src/schema/api-schema.yaml (hunk 7 above)
Comment (suhailkakar) [reply packages/api/src/schema/api-schema.yaml]: oh, sorry! updated it!
